eth_typing/abi.py

```diff
@@ -170,7 +170,50 @@ class ABIFunctionInfo(TypedDict, total=False):
     """Function input parameters."""
 
 
-ABIElement = Union[ABIFunction, ABIConstructor, ABIFallback, ABIReceive, ABIEvent]
+class ABIErrorComponent(TypedDict, total=False):
```

Comment:
It looks like `ABIErrorComponent` and `ABIErrorParam` are exactly the same. Is there a conflict in creating one of these types and nesting it?

Reply:
I can give that a try, this was all similar to ABIFunction/Param/Component and I wasnt sure what the recursive typing was doing there. If this works I can update those as well.

Reply:
I see. If not, it may be worth creating `ABIComponent` and `ABIParam` and using it in both `ABIError` and `ABIFunction`. I don't see a reason for `ABIFunctionParam`, `ABIFunctionComponent`, `ABIErrorParam`, and `ABIErrorComponent`... I can see this spiraling off into other instances as well and that just doesn't seem ideal. Since we get to re-write this, let's try to be as efficient as we can if it makes sense.